### PR TITLE
Task 6: POST comment to specified article and related error handling

### PR DESCRIPTION
**Pull request by jcampb28**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -165,4 +165,55 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+    .expect(201)
+    .then((response) => {
+      expect(response.body).toEqual({
+        comment_id: 19,
+        votes: 0,
+        article_id: 2,
+        author: "butter_bridge",
+        body: "What a nice article",
+        created_at: expect.any(String)
+      });
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:185]: A great test that is specific where possible and vague where necessary
I would perhaps make the `comment_id` `expect.any(Number)`, just in case you ever change or add to your seed data, to make it a bit more flexible.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -165,4 +165,55 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+    const newComment = {username: "butter_bridge", body: ""}
+    return request(app)
+    .post("/api/articles/2/comments")
+    .send(newComment)
+    .expect(400)
+    .then((response) => {
+      expect(response.body.msg).toBe("Bad Request")
+    });
+  });
+  test("400: Responds with a bad request error if the username is an empty string", () => {
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:198]: There's a case for this being a 404 - username does not exist

File: __tests__/app.test.js [hunk 3]
```diff
@@ -165,4 +165,55 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+  });
+  test("400: Responds with a bad request error if the comment is an empty string", () => {
+    const newComment = {username: "butter_bridge", body: ""}
+    return request(app)
+    .post("/api/articles/2/comments")
+    .send(newComment)
+    .expect(400)
+    .then((response) => {
+      expect(response.body.msg).toBe("Bad Request")
+    });
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:196]: It would be good to add a 400 test that check what happens when a comment object without the correct keys is used, an empty object for example.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -165,4 +165,55 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+  test("404: Responds with a not found error if the comment post request is being made to a non-existent article", () => {
+    const newComment = {username: "butter_bridge", body: "What a nice article"}
+    return request(app)
+    .post("/api/articles/256/comments")
+    .send(newComment)
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe("No article with specified ID found")
+    });
+  });
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:217]: Great error handling test for a non-existent article id, would also be good to add a test that checks you get a 400 when an invalid article id is provided
For example `/api/articles/katherine/comments`

File: app/controllers/articles.controller.js [hunk 5]
```diff
@@ -1,26 +1,35 @@
...
-        res.status(200).send(comments)
+        res.status(200).send(comments);
     }).catch(next);
 };
 
-module.exports = { getArticlesById, getArticles, getArticleComments }
\ No newline at end of file
+const postCommentToArticle = (req, res, next) => {    
+    const {username, body} = req.body;
+    const {article_id} = req.params;
```
Comment (nc-katherine-hurst) [inline app/controllers/articles.controller.js:28]: Cool use of destructuring 😎

File: app/controllers/articles.controller.js [hunk 6]
```diff
@@ -1,26 +1,35 @@
...
+    addCommentToArticle(username, body, article_id)
+    .then((comment) => {
+        res.status(201).send(comment);
```
Comment (nc-katherine-hurst) [inline app/controllers/articles.controller.js:31]: It would be best to here (and throughout) follow the best practise of returning your data in an object on a semantic key
So for example `{ comment: { <your comment object> } }`

**Review by nc-katherine-hurst — COMMENTED** [review]
Nice work Jennifer, please find your feedback below 🙂
